gpii/node_modules/flowManager/src/BrowserChannel.js

```diff
@@ -0,0 +1,53 @@
+(function () {
+
+    "use strict";
+
+    var fluid = require("infusion");
+    var gpii = fluid.registerNamespace("gpii");
+
+    fluid.registerNamespace("gpii.request.flowManager");
+
+    fluid.defaults("gpii.flowManager.browserChannel", {
+        gradeNames: ["fluid.standardRelayComponent", "autoInit"]
+    });
+
+    fluid.defaults("kettle.requests.request.handler.browserChannel", {
+        gradeNames: ["gpii.request.flowManager.sessionAware", "autoInit"],
+        invokers: {
+            handle: {
+                funcName: "gpii.request.flowManager.browserChannelHandle",
+                args: ["{request}", "{requestProxy}.events",
+                       "{flowManager}.solutionsRegistryDataSource",
+                       "{deviceReporter}.platformReporter"],
+                dynamic: true
+            }
+        }
+    });
+
+    gpii.request.flowManager.browserChannelHandle = function (fluidRequest, events, solutionsRegistryDataSource, platformReporter) {
+        solutionsRegistryDataSource.get({os: platformReporter.reportPlatform().id}, function onSuccess (entries) {
+            var solutionId = fluidRequest.data;
+
+            if (!(solutionId in entries)) {
+                var msg = "Rejecting a connection request from '" + solutionId +
+                          "'. The solution id was not found in the solutions registry";
+
+                fluidRequest.socket.emit("error", msg);
```

Comment:
Presumably the logAndNotify line already does this - so in theory both of these lines could be removed